src/Microsoft.IdentityModel.Tokens/Validation/Results/ValidatedToken.cs

```diff
@@ -46,29 +37,61 @@ public void Log(ILogger logger)
                 ValidatedSigningKey?.KeyId ?? "none",
                 ActorValidationResult is not null
             );
-        }
 
-        public SecurityToken SecurityToken { get; private set; }
+        /// <summary>
+        /// The <see cref="SecurityToken"/> that was validated.
+        /// </summary>
+        public SecurityToken SecurityToken { get; private set; } = securityToken ?? throw new ArgumentNullException(nameof(securityToken));
 
-        public TokenHandler TokenHandler { get; private set; }
+        /// <summary>
+        /// The <see cref="TokenHandler"/> that was used to validate the token.
+        /// </summary>
+        public TokenHandler TokenHandler { get; private set; } = tokenHandler ?? throw new ArgumentNullException(nameof(tokenHandler));
 
-        public ValidationParameters ValidationParameters { get; private set; }
+        /// <summary>
+        /// The <see cref="ValidationParameters"/> that were used to validate the token.
+        /// </summary>
+        public ValidationParameters ValidationParameters { get; private set; } = validationParameters ?? throw new ArgumentNullException(nameof(validationParameters));
 
         #region Validated Properties
+        /// <summary>
+        /// The result of validating the actor, if any.
+        /// </summary>
         public ValidatedToken? ActorValidationResult { get; internal set; }
 
+        /// <summary>
+        /// The audience that was validated, if any.
+        /// </summary>
         public string? ValidatedAudience { get; internal set; }
 
+        /// <summary>
+        /// The issuer that was validated. If present, it contains the source of the validation as well.
+        /// </summary>
         public ValidatedIssuer? ValidatedIssuer { get; internal set; }
```

Comment:
Is the idea that the 'internal set' will become public so the user can set the value?